python/cuml/dask/linear_model/base.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.dask.common import wait_and_raise_from_futures
+from cuml.dask.common.comms import CommsContext
+from cuml.dask.common.input_utils import DistributedDataHandler
+
+
+class BaseLinearModelSyncFitMixin(object):
+
+    def _fit(self, model_func, data, **kwargs):
+
+        n_cols = data[0].shape[1]
+
+        data = DistributedDataHandler.create(data=data, client=self.client)
+        self.datatype = data.datatype
+
+        comms = CommsContext(comms_p2p=False, verbose=self.verbose)
+        comms.init(workers=data.workers)
+
+        data.calculate_parts_to_sizes(comms)
+        self.ranks = data.ranks
+
+        lin_models = dict([(data.worker_info[wf[0]]["rank"],
+                            self.client.submit(
+            model_func,
+            comms.sessionId,
+            self.datatype,
+            **self.kwargs,
+            pure=False,
+            workers=[wf[0]]))
+            for idx, wf in enumerate(data.worker_to_parts.items())])
+
+        lin_fit = dict([(wf[0], self.client.submit(
+            _func_fit,
+            lin_models[data.worker_info[wf[0]]["rank"]],
+            wf[1],
+            data.total_rows,
+            n_cols,
+            data.parts_to_sizes[data.worker_info[wf[0]]["rank"]],
+            data.worker_info[wf[0]]["rank"],
+            pure=False,
+            workers=[wf[0]]))
+            for idx, wf in enumerate(data.worker_to_parts.items())])
```

Comment:
I found the wf[0] everywhere hard to parse... could you change it to something with a name using unpacking like `for idx, (worker, part) in enumerate...`?

Reply:
I will address this in https://github.com/rapidsai/cuml/issues/2414